go-selinux/selinux_linux.go

```diff
@@ -250,8 +250,17 @@ func getSELinuxPolicyRoot() string {
 }
 
 func isProcHandle(fh *os.File) error {
-	var buf unix.Statfs_t
-	err := unix.Fstatfs(int(fh.Fd()), &buf)
+	var (
+		buf unix.Statfs_t
+		err error
+	)
+
+	for {
+		err = unix.Fstatfs(int(fh.Fd()), &buf)
+		if err != unix.EINTR {
```

Comment:
I wonder if the approach taken in `verifySELinuxfsMount()` above is somewhat clearer. (I realise it does the same, wondering if that approach causes head-scratching  for the occasional reader).

```go
for {
	err = unix.Fstatfs(int(fh.Fd()), &buf)
	if err == nil {
			break
	}
	if err != unix.EINTR {
		return errors.Wrapf(err, "statfs(%q) failed", fh.Name())
	}
}
```

Or somewhat shorter (not sure if clearer)

```go
for err := unix.Fstatfs(int(fh.Fd()), &buf); err != nil; {
	if err == unix.EINTR {
		continue
	}
	return errors.Wrapf(err, "statfs(%q) failed", fh.Name())
}
```

Reply:
Seems that the second version won't work, but the first one, yes, it looks better than the one I wrote this time.

Updated; PTAL